frontend/src/components/custom-tools/combined-output/CombinedOutput.jsx

```diff
@@ -111,16 +133,44 @@ function CombinedOutput({ docId, setFilledFields }) {
       });
   };
 
+  const getAdapterInfo = () => {
+    axiosPrivate
+      .get(`/api/v1/unstract/${sessionDetails.orgId}/adapter/?adapter_type=LLM`)
+      .then((res) => {
+        const adapterList = res.data;
+        setAdapterData(getLLMModelNamesForProfiles(llmProfiles, adapterList));
+      });
+  };
+
   if (isOutputLoading) {
     return <SpinnerLoader />;
   }
 
+  const handleTabChange = (key) => {
+    if (key === "0") {
+      setSelectedProfile(defaultLlmProfile);
+    } else {
+      setSelectedProfile(adapterData[key - 1]?.profile_id);
+    }
+    setActiveKey(key);
+  };
+
   return (
     <div className="combined-op-layout">
       <div className="combined-op-header">
+        <Tabs activeKey={activeKey} onChange={handleTabChange} moreIcon={<></>}>
+          <TabPane tab={<span>{"Default"}</span>} key={"0"}></TabPane>
+          {adapterData.map((adapter, index) => (
+            <TabPane
+              tab={<span>{adapter.llm_model}</span>}
+              key={(index + 1)?.toString()}
+            ></TabPane>
```

Comment:
@jagadeeswaran-zipstack NIT: We can make this a self closing tag.

```suggestion
            <TabPane
              tab={<span>{adapter.llm_model}</span>}
              key={(index + 1)?.toString()}
            />
```